src/components/_auth/SignInSteps/CreateMemberModal.tsx

```diff
@@ -1,53 +1,89 @@
 import { useApolloClient } from '@apollo/client'
+import { zodResolver } from '@hookform/resolvers/zod'
 import debouncePromise from 'awesome-debounce-promise'
 import axios, { AxiosError } from 'axios'
 import React, { useEffect, useRef, useState } from 'react'
 import { useForm } from 'react-hook-form'
 import { useNavigate } from 'react-router'
+import * as z from 'zod'
 
 import { useQueryNodeStateSubscription } from '@/api/hooks'
 import { GetMembershipDocument, GetMembershipQuery, GetMembershipQueryVariables } from '@/api/queries'
+import { Text } from '@/components/Text'
 import { ViewErrorFallback } from '@/components/ViewErrorFallback'
 import { TextArea } from '@/components/_inputs/TextArea'
 import { Loader } from '@/components/_loaders/Loader'
 import { MEMBERSHIP_NAME_PATTERN, URL_PATTERN } from '@/config/regex'
-import { absoluteRoutes } from '@/config/routes'
+import { QUERY_PARAMS } from '@/config/routes'
 import { FAUCET_URL } from '@/config/urls'
+import { useRouterQuery } from '@/hooks/useRouterQuery'
 import { MemberId } from '@/joystream-lib'
 import { useConfirmationModal } from '@/providers/confirmationModal'
 import { useConnectionStatusStore } from '@/providers/connectionStatus'
+import { useSnackbar } from '@/providers/snackbars'
 import { useUser } from '@/providers/user'
-import { textFieldValidation } from '@/utils/formValidationOptions'
 import { SentryLogger } from '@/utils/logs'
 
-import {
-  Form,
-  Header,
-  Hero,
-  StyledAvatar,
-  StyledButton,
-  StyledTextField,
-  SubTitle,
-  Wrapper,
-} from './CreateMemberView.styles'
+import { StyledAvatar, StyledButton, StyledDialogModal, StyledTextField, Wrapper } from './CreateMemberModal.styles'
 
 type Inputs = {
   handle: string
   avatar: string
   about: string
 }
 
-export const CreateMemberView = () => {
-  const { activeAccountId, refetchMemberships } = useUser()
+export const CreateMemberModal: React.FC = () => {
+  const { activeAccountId, refetchMemberships, extensionConnected, setActiveUser } = useUser()
   const nodeConnectionStatus = useConnectionStatusStore((state) => state.nodeConnectionStatus)
-
+  const step = useRouterQuery(QUERY_PARAMS.LOGIN)
   const navigate = useNavigate()
+
+  const schema = z.object({
```

Comment:
This uses some of the refs that are defined further down, maybe it would be clearer to read if we moved the schema definition after all the refs?